packages/dynamite/lib/src/openapi_builder.dart

```diff
@@ -1268,7 +1268,9 @@ TypeResult resolveObject(
                     final isContentStringArray = isContentString && result is TypeResultList;
                     final hasExtraJsonKeyValues =
                         extraJsonKeyValues != null && extraJsonKeyValues.containsKey(propertyName);
-                    if (hasDifferentName || isContentString || isContentStringArray || hasExtraJsonKeyValues) {
+                    if (hasDifferentName ||
+                        result is! TypeResultBase && (isContentString || isContentStringArray) ||
```

Comment:
You need to put parenthesis around `result is! TypeResultBase && (isContentString || isContentStringArray)`